### DISCUSSION TITLE
Longer (or no) debounce delay when editing order adjustment

### DISCUSSION THREAD

**Discussion by elivz** [craftcms/commerce#2476] [Ideas]
**What are you trying to do?**
This is similar to https://github.com/craftcms/commerce/blob/main/src/web/assets/commerceui/src/js/order/components/details/Adjustment.vue. When editing an order adjustment, if you stop typing for more than a second it triggers an AJAX order refresh. However, you can continue to type while that happens, and whatever you have entered gets lost when the refresh finishes.

**What's your proposed solution?**
I would suggest either not auto-saving those changes, and just let the user click the "Update Order" button to persist their changes, or switch to saving onblur instead. One second is a short enough delay that it gets triggered frequently without even typing particularly slowly and if you are not paying attention it is very easy to have half your adjustment description get deleted out from under you.

**Comment by pdaleramirez** [comment]
I was able to replicate this issue but rarely cause the response time is fast. I can see how can this be problematic on orders with slow response time. Perhaps because of an order update event.
